scanpy/preprocessing/normalization.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+from scipy.sparse import issparse
+from sklearn.utils import sparsefuncs
+from .. import logging as logg
+
+def _normalize_data(X, counts, after=None, copy=False):
+    X = X.copy() if copy else X
+    after = np.median(counts) if after is None else after
+    counts += (counts == 0)
+    counts /= after
+    if issparse(X):
+        X = sparsefuncs.inplace_row_scale(X, 1/counts)
+    else:
+        X /= counts[:, None]
+    return X if copy else None
+
+def normalize_quantile(data, counts_per_cell_after=None, counts_per_cell=None,
```

Comment:
Can we have a better name than `counts_per_cell_after`? Maybe `counts_after` or  `normed_value` or `target_norm` or `target_sum` or something? What's the best name for this?

Can we remove the `counts_per_cell` parameter? Previously, it was just there to enable the recursive call.

Reply:
But the new functions need to wrap the old, no? I left it for wrappers.

Reply:
But the new functions need to wrap the old, no? I left it for wrappers.